src/main/java/org/opentripplanner/common/StreetUtils.java

```diff
@@ -25,7 +25,16 @@
     private static Logger LOG = LoggerFactory.getLogger(StreetUtils.class);
     private static int islandCounter = 0;
 
-    public static void pruneFloatingIslands(Graph graph, int maxIslandSize, 
+    /* Island pruning strategy:
+       1. Extract islands without using noThruTraffic edges at all
+       2. Then create expanded islands by accepting noThruTraffic edges, but do not jump across original islands!
+          Note: these expanded islands can overlap.
+       3  Relax connectivity even more: generate islands by allowing jumps between islands. Find out unreachable edges of small islands.
+       4. Analyze small expanded islands (from step 2). Convert edges which are reachable only via noThruTraffic edges
+          to noThruTraffic state. Depedestrianize unreachable edges. Removed unconnected edges.
+     */
+
+    public static void pruneFloatingIslands(Graph graph, int maxIslandSize,
```

Comment:
I notice these methods are only used in PruneFloatingIslands, so perhaps they should be moved to that class.

Also, would it be possible to only do the new behavior by configuration, so OTP users could turn it off if they want? Alternatively, there could be different modules (something like `PruneFloatingIslands` and `PruneNoThruTrafficIslands`) and advanced users could simply add whichever they want in GraphBuilder.